src/Value/Service.php

```diff
@@ -0,0 +1,256 @@
+     * - draft (0/1) : Has the service the draft status.
+     * - countChannels (int) : The number of channels linked to the service.
+     *
+     * @inheritdoc
+     *
+     * @return \StadGent\Services\OpeningHours\Value\Service
+     *
+     * @throws \InvalidArgumentException
+     *   If the created_at/update_at are empty.
+     */
+    public static function fromArray(array $data)
+    {
+        $service = new static();
+
+        if (!empty($data['id'])) {
+            $service->id = (int) $data['id'];
+        }
+        if (!empty($data['uri'])) {
+            $service->uri = $data['uri'];
+        }
+        if (!empty($data['label'])) {
+            $service->label = $data['label'];
+        }
+        if (!empty($data['description'])) {
+            $service->description = $data['description'];
+        }
+        if (!empty($data['countChannels'])) {
+            $service->countChannels = (int) $data['countChannels'];
+        }
+
+        $service->source = ServiceSource::fromArray($data);
+        $service->dateAttributes = DateAttributes::fromArray($data);
+        $service->isDraft = new Boolean(!empty($data['source']));
+
+        return $service;
+    }
+
+
+    /**
+     * Get the unique identifier for the Service.
+     *
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * Get the URI of the service.
+     *
+     * @return string
+     */
+    public function getUri()
+    {
+        return $this->uri;
+    }
+
+    /**
+     * Get the service label.
+     *
+     * @return string
+     */
+    public function getLabel()
+    {
+        return $this->label;
+    }
+
+    /**
+     * Get the service description.
+     *
+     * @return string
+     */
+    public function getDescription()
+    {
+        return $this->description;
+    }
+
+    /**
+     * Get the date-time the service was created.
+     *
+     * @return \StadGent\Services\OpeningHours\Value\DateAttributes
+     */
+    public function getDateAttributes()
+    {
+        return $this->dateAttributes;
+    }
+
+    /**
+     * Get the source name (if any) of the service.
+     *
+     * @return \StadGent\Services\OpeningHours\Value\ServiceSource
+     */
+    public function getSource()
+    {
+        return $this->source;
+    }
+
+    /**
+     * Check if the service has a source.
+     *
+     * @return bool
+     */
+    public function hasSource()
+    {
+        return $this->getSource()->getName() !== null
+            && $this->getSource()->getId() !== null;
+    }
+
+    /**
+     * Is the service in draft status.
+     *
+     * @return \StadGent\Services\OpeningHours\Value\Boolean
+     */
+    public function getDraft()
+    {
+        return $this->isDraft;
+    }
+
+    /**
+     * Check if the service is in draft status.
+     *
+     * @return bool
+     */
+    public function isDraft()
+    {
+        return $this->getDraft()->isTrue();
```

Comment:
correct maar verwarrend;

is deze functie wel nodig? zou leuk zijn als je een Boolean object kon casten naar een bool (zie https://github.com/p1ncet/obcast)
misschien is het zo duidelijker dat isDraft een object is: $this->isDraft->isTrue();